itests/org.openhab.core.thing.tests/src/main/java/org/openhab/core/thing/binding/BindingBaseClassesOSGiTest.java

```diff
@@ -606,6 +611,36 @@ public void assertConfigurationParametersAreValidated() {
                 () -> thingRegistry.updateConfiguration(thingUID, Map.of("parameter", configuration)));
     }
 
+    @Test
+    public void assertIllegalConfigurationParametersPreventUpdate() {
+        SimpleThingHandlerFactory thingHandlerFactory = new SimpleThingHandlerFactory();
+        thingHandlerFactory.activate(componentContext);
+        registerService(thingHandlerFactory, ThingHandlerFactory.class.getName());
+
+        registerThingTypeProvider();
+        registerConfigDescriptionProvider(true);
+
+        ThingTypeUID thingTypeUID = new ThingTypeUID("bindingId:type");
+        ThingUID thingUID = new ThingUID("bindingId:type:thingId");
+        Thing thing = ThingBuilder.create(thingTypeUID, thingUID)
+                .withConfiguration(new Configuration(Map.of("parameter", "someValue"))).build();
+
+        managedThingProvider.add(thing);
+
+        SimpleThingHandler handler = (SimpleThingHandler) thing.getHandler();
+        assertNotNull(handler);
```

Comment:
Should we add an initial assertion for the parameter equals to `"some value"`?

Reply:
done